src/main/java/org/broadinstitute/hellbender/tools/gvs/common/WeightedSplitIntervals.java

```diff
@@ -0,0 +1,219 @@
+        float targetWeightPerScatter = totalWeight / (float) scatterCount;
+
+        System.out.println("Total Weight:"+totalWeight + " scatterCount: " + scatterCount + " target:" + targetWeightPerScatter);
+
+        final int maxNumberOfPlaces = Math.max((int)Math.floor(Math.log10(scatterCount-1))+1, numDigits);
+        final String formatString = "%0" + maxNumberOfPlaces + "d";
+
+        int scatterPiece = 0;
+        float cumulativeWeight = 0;
+        IntervalList currentList = new IntervalList(sequenceDictionary);
+        String lastContig = null;
+
+        PushbackIterator<Interval> iter = new PushbackIterator<>(intervalsWithWeights.iterator());
+        while (iter.hasNext()) {
+            WeightedInterval wi = (WeightedInterval) iter.next();
+
+            // if we're not mixing contigs, but we switched contigs, emit the list
+            if (dontMixContigs && lastContig != null && !lastContig.equals(wi.getContig()) ) {
+                // write out the current list (uniqued and sorted) and start a new one
+                currentList.uniqued().sorted().write(new File(outputDir, prefix + String.format(formatString, scatterPiece++) + extension));
+                currentList = new IntervalList(sequenceDictionary);
+                cumulativeWeight = 0;
+            }
+
+            // in either case, set the last seen contig
+            lastContig = wi.getContig();
+
+            // if the interval fits completely, just add it
+            if (cumulativeWeight + wi.getWeight() <= targetWeightPerScatter ) {
+                cumulativeWeight += wi.getWeight();
+                currentList.add(wi);
+
+            // if it would push us over the edge
+            } else {
+                // add a piece of it
+                float remainingSpace = targetWeightPerScatter - cumulativeWeight;
+
+                // how many bases can we take?
+                int basesToTake = (int) Math.floor(remainingSpace /  wi.getWeightPerBase());
+
+                // split and add the first part into this list
+                WeightedInterval[] pair = wi.split(basesToTake);
+                currentList.add(pair[0]);
+
+                // push the remainder back onto the iterator
+                iter.pushback(pair[1]);
+
+                // add uniqued, sorted output list and reset
+                currentList.uniqued().sorted().write(new File(outputDir, prefix + String.format(formatString, scatterPiece++) + extension));
+                currentList = new IntervalList(sequenceDictionary);
+                cumulativeWeight = 0;
+            }
+        }
+        // write the final list
+        currentList.uniqued().sorted().write(new File(outputDir, prefix + String.format(formatString, scatterPiece++) + extension));
+    }
+
+    @Override
+    public void traverse() { }  // no traversal for this tool!
+
+    protected IntervalList preprocessIntervalsWithWeights(SAMSequenceDictionary sequenceDictionary, Collection<SimpleInterval> intervals) {
```

Comment:
I would restructure a this a bit.  I think it would make sense to pull out a method which converts the file name -> overlapdetector, and pass that as a parameter into this.  Then you could separate the loading the file from the processing of the intervals.  I'm also just partial to making everything with logic static if you can.  

I generally avoid protected unless I'm trying to signal that it's intended to be overriden which I  don't really think this is.  You can use default scope if it needs to be visible for testing, otherwise private is probably best.

Reply:
happy to follow the GATK philosophy here

Reply:
Actually -- I think it's a good idea to factor out the creation of the OverlapDetector to a method, but it's better to encapsulate it here.  The OD is just an implementation detail of how the weights are added to the input intervals.  It's not used for anything else, so encapsulating it here rather than having the caller be exposed to that seems cleaner